test/offsetarray.jl

```diff
@@ -786,3 +786,8 @@ end
         @test b[i] == a[r[i]]
     end
 end
+
+@testset "proper patition for non-1-indexed vector" begin
+    @test partition(OffsetArray(1:10,10), 5) |> collect == [1:5,6:10] # OffsetVector
```

Comment:
and also `partition(OffsetArray(collect(1:10),10), 5) |> collect`. It currently sigfaults with improper index.